### PR TITLE
CW V2 Dashboard CRD operations

### PR DESCRIPTION
**Pull request by pinzon**
## Motivation
Add the operations related to `Dashboard` for cloudwatch v2 provider

## Changes

- [x] PutDashboard
- [x] GetDashboard
- [x] DeleteDashboard
- [x] ListDashboards

## Testing
The test should pass with  Cloudwatch v2 pipeline

### CODE REVIEW

File: localstack/services/cloudwatch/provider_v2.py [hunk 1]
```diff
@@ -214,3 +221,46 @@ def tag_resource(
...
+            dashboard_name, account_id=context.account_id, region_name=context.region
+        )
+        if not dashboard_body:
+            raise InvalidParameterValueException(f"Dashboard {dashboard_name} does not exist.")
+        return GetDashboardOutput(
+            DashboardName=dashboard_name, DashboardBody=dashboard_body, DashboardArn=dashboard_arn
+        )
+
+    def delete_dashboards(self, context: RequestContext, dashboard_names: DashboardNames) -> None:
+        store = self.get_store(context.account_id, context.region)
```
Comment (steffyP) [inline localstack/services/cloudwatch/provider_v2.py:247]: NIT: does it throw an error when we try to delete a dashboard that does not exist?

File: localstack/services/cloudwatch/provider_v2.py [hunk 2]
```diff
@@ -214,3 +221,46 @@ def tag_resource(
...
+    def list_dashboards(
+        self,
+        context: RequestContext,
+        dashboard_name_prefix: DashboardNamePrefix = None,
+        next_token: NextToken = None,
+    ) -> ListDashboardsOutput:
+        store = self.get_store(context.account_id, context.region)
+        dashboard_names = list(store.Dashboards.keys())
+        dashboard_name_prefix = dashboard_name_prefix or ""
+        if dashboard_name_prefix:
```
Comment (steffyP) [inline localstack/services/cloudwatch/provider_v2.py:260]: NIT: maybe we could enhance the existing dashboard test, to also include a `list_dashboards` with a dashboard_name_prefix filter? Then we could also add a snapshot for this one :)

**Review by steffyP — APPROVED** [review]
Nice work, @pinzon! 🚀 
Changes LGTM, sorry for the hick-up with the rebase. 
Let's wait for the tests before merging.

I added two improvement suggestions :)  you can either fix them with this PR, or add it to the list of follow up tasks in #9347 🙏

File: localstack/services/cloudwatch/provider_v2.py (hunk 1 above)
Comment (pinzon) [reply localstack/services/cloudwatch/provider_v2.py:247]: No. it doesn't

**Pull request merged**